### DISCUSSION TITLE
Some files are owned by different user in bind mount

### DISCUSSION THREAD

**Discussion by KShivum** [containers/podman#22005] [Q&A]
So I'm having trouble moving over from Docker to Podman. I have a Gitlab docker compose file and when running it, most of the files are the user that ran the compose but it seems like some data and the redis server are some other user/group, is it possible to have it all as one user?
https://www.redhat.com/sysadmin/container-permission-denied-errors
I was following this and I found the :U suffix for the bind mount, but it doesn't seem like it works in compose

**Comment by eriksjolund** [comment]
You can use `--userns keep-id:uid=$uid,gid=$gid` to map a container user to your regular user on the host.
Files that this container user creates will be owned by your regular user on the host.

The tricky part can be to know which $uid and $gid to use. I tried to automate this detection with a script:
https://github.com/eriksjolund/podman-detect-option

For __docker.io/library/redis__ it seems you would need to use 
 `--userns keep-id:uid=999,gid=999`
or maybe
 `--userns keep-id:uid=999,gid=999 --user 999:999`

**Reply by eriksjolund** [reply]
I forgot to mention:
I don't know if  there is a way to provide the `--userns keep-id:uid=999,gid=999` option to docker-compose.
An alternative is to migrate the compose file to quadlet files, for instance with [podlet](https://github.com/k9withabone/podlet)

**Reply by KShivum** [reply]
So the issue is, the gitlab ce image only uses one container, I don't have access to the redis instance it makes, I tried to set the id to my user, however gitlab itself kept getting permission denied errors. I'll see if I can post my compose file a bit later
